src/Stratis.Bitcoin/P2P/PeerDiscoveryLoop.cs

```diff
@@ -106,47 +114,57 @@ public void DiscoverPeers(IConnectionManager connectionManager)
         }
 
         /// <summary>
-        /// See <see cref="DiscoverPeers"/>.
+        /// See <see cref="DiscoverPeers"/>. This loop deals with discovery from DNS seeds and seed nodes as opposed to peers.
         /// </summary>
-        private async Task DiscoverPeersAsync()
+        private async Task DiscoverFromDnsSeedsAsync()
         {
             var peersToDiscover = new List<IPEndPoint>();
-            List<PeerAddress> foundPeers = this.peerAddressManager.PeerSelector.SelectPeersForDiscovery(1000).ToList();
-            peersToDiscover.AddRange(foundPeers.Select(p => p.Endpoint));
 
-            if (peersToDiscover.Count == 0)
+            if (this.peerAddressManager.Peers.Count == 0 || !this.peerAddressManager.Peers.Select(a => !a.Attempted).Any())
```

Comment:
@zeptin what is this for again?

`!this.peerAddressManager.Peers.Select(a => !a.Attempted).Any()`?

Reply:
Perhaps change this `if` so that we return with a log message if there are peers in the address manager:

"DISCOVERY_SKIPPED_ADDRESSBOOK_NOT_EMPTY".

And then the same thing with the second condition, log and return.

Reply:
> @zeptin what is this for again?
> 
> `!this.peerAddressManager.Peers.Select(a => !a.Attempted).Any()`?

This checks if there are no peers in the manager that have not been attempted yet, as that also means we should try the seeds again to try get more options.